include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,81 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
+  TritonGPU_Attr<"SwizzledBlocksSharedEncoding", "swizzled_blocks_shared_encoding", [SharedEncodingTrait, LayoutEncodingTrait]> {
+  let mnemonic = "swizzled_blocks_shared";
+
+  let description = [{
+This shared encoding is similar to SwizzledSharedEncodingAttr,
+but instead of repeating swizzling pattern every `blockSize = maxPhase*perPhase` rows of memory object,
+this layout changes swizzling pattern.
```

Comment:
but instead of repeating swizzling pattern every `maxPhase * perPhase` row,
called a block, this layout changes to use a different block size.

Reply:
Right, this pattern has multiple nested levels of "blocks", I just decided to name one of them to use in variable names. This comment explicitly defines block as maxPhase*perPhase rows, so I think it should be good enough for later usage.

I don't have a better name for this entity at this moment, you can suggest another if you want.

Reply:
Oh, my previous comment was suggesting another wording; not a question. I should put them in quotes actually like:

".. but instead of repeating swizzling pattern every maxPhase * perPhase row, called a block, this layout changes to use a different block size."

I see Lixun has better suggestions regarding the terms though.